### ISSUE TITLE
MLFlow Crashes when Logging NaN Param

### ISSUE THREAD

**Issue by jakerachleff-databricks** [mlflow/mlflow#1234]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Databricks
- **MLflow installed from (source or binary)**: 
Unsure
- **MLflow version (run ``mlflow --version``)**:
0.9.1
- **Python version**: 
Unsure
- **npm version (if running the dev UI):
- **Exact command to reproduce**:
`mlflow.log_param("param", np.nan)`

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in MLflow or a feature request.

MLFlow UX Issue:

When running hyperparameter tuning with MLflow, an error was produced when my training loss was NaN. Specifically, the follow exception was risen:

`RestException: INVALID_PARAMETER_VALUE: Metric 'Epoch Training Loss' cannot have value NaN`

This seems like errant behavior, since I'd ideally like to hyperparm tune over many combinations of params, and thus not have my tuning cut off from an accidental NaN. It seems like NaNs may occur in this process, and the user should not be responsible for checking for them. 

Ideally, this would just warn the user, instead of halting their tuning.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks,
please include the full traceback. Large logs and files should be attached.
Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by mateiz** [comment]
That's a good catch.. we need to decide what we want to do here.

**Comment by andrewmchen** [comment]
Seems to only be able to be repro-ed on Databricks.

**Comment by ankitmathur-db** [comment]
@tomasatdatabricks we closed this right?

**Comment by tomasatdatabricks** [comment]
Yes, you can log NaNs and Infs with mlflow>=1.1.
